File: Easy/happy_numbers.py
data = ['1', '7', '22']

def happy_numbers(n):
	D = dict()
	n = int(n)
	while n != 1:
		n = str(n)
		n = list(n)
		n = map(int, n)
		n2 = [i**2 for i in n]
		n = sum(n2)
		if n in D:
			return 0
		else:
			D[n] = 1
		
	
	return 1

#import sys
#test_cases = open(sys.argv[1], 'r')
#data = test_cases.readlines()
for test in data:
	H = happy_numbers(test)
	print H

#test_cases.close()

File: Easy/happy_numbers.r
test.cases <- c("1", "7", "22")

happy <- function(x, visited){
			visited <<- c()
			n <- as.numeric(strsplit(as.character(x), "")[[1]])
			n <- sum(n^2)
			if (n == 1){
				TRUE
			}
			else if (n %in% visited) {
				FALSE
			} else {
				visited <- c(visited, n)
				happy(n, visited)
			}
}

#args <- commandArgs(trailingOnly=TRUE)
#test.cases <- strsplit(readLines(args[[1]], warn=FALSE), '\n')
cat(sapply(test.cases, function(test) {
	if (length(test) > 0) {
		num <- as.numeric(test)
		visited <- c()
		if (happy(num, visited)) {
			1
		} else {
			0
		}
	}
}), sep = "\n")